src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/SemanticRange.cs

```diff
@@ -29,20 +39,27 @@ public SemanticRange(int kind, Range range, int modifier, bool fromRazor)
     /// </summary>
     public bool FromRazor { get; }
 
-    public int CompareTo(SemanticRange? other)
+    public int CompareTo(SemanticRange other)
     {
-        if (other is null)
+        var startCompare = Range.StartCharacter.CompareTo(other.Range.StartCharacter);
+        if (startCompare != 0)
         {
-            return 1;
+            return startCompare;
         }
 
-        var startCompare = Range.Start.CompareTo(other.Range.Start);
+        startCompare = Range.StartLine.CompareTo(other.Range.StartLine);
```

Comment:
Compare line before character. Line 5, Column 0 should not come before Line 1, Column 5. Otherwise this in general looks fine